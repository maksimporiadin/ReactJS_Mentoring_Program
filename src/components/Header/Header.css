.header {
    position: fixed;
    background: #414141;
    width: 100%;
    margin: 0 auto;
    padding: 15px;
}

.header-logo {
    margin: 0 50px;
    justify-content: space-between;
    display: flex;
}

.header-search-button {
    border-radius: 0;
    color: red;
    background: white;
    padding: 5px 20px;
    font-weight: bold;
    text-transform: uppercase;
}