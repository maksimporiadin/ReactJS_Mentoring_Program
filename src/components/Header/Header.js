import React, { Component } from 'react';

import { Button, Logo } from '../UI';

import './header.css';

export const Header = ({ isShowSearchButton, children }) => (
    <header className="header">
        <div className="header-logo">
            <Logo/>
            {
                isShowSearchButton &&
                <Button
                    className="btn-primary header-search-button"
                    onClick={() => window.location = '/'} >
                    Search
                </Button>
            }
        </div>
        {children}
    </header>
);
