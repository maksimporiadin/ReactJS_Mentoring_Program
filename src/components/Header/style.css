.header {
    /*position: fixed;*/
    background: #414141;
    width: 100%;
    margin: 0 auto;
    padding: 15px 50px;
}

.header-logo {
    justify-content: space-between;
    display: flex;
}

.header-search-button {
    line-height: 0;
}