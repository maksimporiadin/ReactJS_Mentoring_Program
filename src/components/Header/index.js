import React, { Component } from 'react';
import { Redirect  } from 'react-router';

import { Button, Logo } from '../UI';

import './style.css';

class Header extends Component {
    state = {
        isClicked: false
    }

    render () {
        return (
            <header className="header">
                {
                    this.state.isClicked &&
                    <Redirect push to="/movies" />
                }
                <div className="header-logo">
                    <Logo/>
                    {
                        this.props.isShowSearchButton &&
                        <Button
                            className="btn-primary header-search-button"
                            onClick={() => this.setState({ isClicked: true })} >
                            Search
                        </Button>
                    }
                </div>
                {this.props.children}
            </header>
        );
    }
}

export default Header;
