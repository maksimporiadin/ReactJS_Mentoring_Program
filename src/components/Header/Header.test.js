import 'jsdom-global/register';
import React from 'react';

import Header from './index';

import Adapter from 'enzyme-adapter-react-16';
import { shallow, configure } from 'enzyme';

configure({adapter: new Adapter()});

describe('Header', () => {
    let wrapper;

    beforeEach(()=>{
        jest.resetAllMocks();
        wrapper = shallow(<Header />)
    });

    it('is renders', () => {
        expect(wrapper).toHaveLength(1);
    });

    it('HandlerOnClick test', () => {
        wrapper.instance().HandlerOnClick();
        expect(wrapper.state('isClicked')).toBe(true);
    });
});