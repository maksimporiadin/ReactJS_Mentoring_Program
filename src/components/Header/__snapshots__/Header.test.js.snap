// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header is renders 1`] = `
<header
  className="header"
>
  <div
    className="header-logo"
  >
    <Logo />
  </div>
</header>
`;
